src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -1234,13 +1235,14 @@ internal bool Compile(bool optimized)
 
         internal static void LogScriptBlockCreation(ScriptBlock scriptBlock, bool force)
         {
-            if (force || ShouldLogScriptBlockActivity("EnableScriptBlockLogging"))
+            ScriptBlockLogging logSetting = GetScriptBlockLoggingSetting();
+            if (force || logSetting?.EnableScriptBlockLogging == true)
             {
                 if (!scriptBlock.HasLogged || InternalTestHooks.ForceScriptBlockLogging)
                 {
                     // If script block logging is explicitly disabled, or it's from a trusted
                     // file or internal, skip logging.
-                    if (ScriptBlockLoggingExplicitlyDisabled() ||
+                    if (logSetting?.EnableScriptBlockLogging == false ||
```

Comment:
There is no behavior change, the check `logSetting?.EnableScriptBlockLogging == false` is to make sure that the `EnableScriptBlockLogging` is explicitly set to `false`. If you look at the original implementation of `ScriptBlockLoggingExplicitlyDisabled()`, it returns `true` only if the registry key `EnableScriptBlockLogging` exists and its value is explicitly 0.

I agree to have docs to clarify how logging works in PowerShell. Not sure if we already have some docs about it.

Reply:
@daxian-dbw Could you please add this in PR description for future docs?

Reply:
If we don't, the new about_Logging.md would be a reasonable place.
As far as the behavior, the implementation currently has three behaviors - 
1: EnableScriptBlockLogging == true: always log scriptblocks, 
2: EnableScriptBlockLogging == false: don't log scripblocks
3: EnableScriptBlockLogging not defined: log suspicious scriptblocks

Reading through this again, the implementation should be simplified: 
The force parameter is explicitly tied to HasSuspiciousContent in the caller so you might as well do the work in LogScriptBlockCreation.
In the current implementation, #2 and #3 are effectively identical since we pass force = true for suspicious script blocks. In short, false and default have the same behavior.

Reply:
Yes, a documentation section is added to PR description.